Source/WTF/wtf/cocoa/UUIDCocoa.mm

```diff
@@ -0,0 +1,46 @@
+/*
+ * Copyright (C) 2022 Apple Inc. All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions
+ * are met:
+ * 1. Redistributions of source code must retain the above copyright
+ *    notice, this list of conditions and the following disclaimer.
+ * 2. Redistributions in binary form must reproduce the above copyright
+ *    notice, this list of conditions and the following disclaimer in the
+ *    documentation and/or other materials provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED BY APPLE INC. AND ITS CONTRIBUTORS ``AS IS''
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO,
+ * THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
+ * PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL APPLE INC. OR ITS CONTRIBUTORS
+ * BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+ * CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ * SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF
+ * THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#import "config.h"
+#import "UUID.h"
+
+#import "RetainPtr.h"
+
+namespace WTF {
+
+RetainPtr<NSUUID> UUID::createNSUUID() const
+{
+    return adoptNS([[NSUUID alloc] initWithUUIDString:toString()]).autorelease();
+}
+
+// Use string instead of bytes to avoid consideration of endianess (NSUUID stores UUID as array of bytes,
+// and we store UUID as UInt128).
+UUID::UUID(NSUUID * nsUUID)
+    : UUID(convertStringToInt(String([nsUUID UUIDString])))
```

Comment:
If you make the change above to revert the interface to parse(), then you can call parse() here and just unconditionally dereference the result, since we know an NSUUID is a valid UUID.

Reply:
Alternatively, if we believe there is some reason NSUUID might sometimes fail to convert to UUID, then we should use an explicit "from(NSUUID *)" interface that returns an optional<UUID>.

Reply:
I think NSUUID is a valid UUID so I will go with option 1.